doc/tools/coqrst/coqdomain.py

```diff
@@ -400,57 +400,101 @@ class TableObject(NotationObject):
 
     Example::
 
-        .. prodn:: term += let: @pattern := @term in @term
         .. prodn:: occ_switch ::= { {? {| + | - } } {* @num } }
+        .. prodn:: term += let: @pattern := @term in @term
+        .. prodn:: | second_production
+
+       The first line defines "occ_switch", which must be unique in the document.  The second
+       references but doesn't define "term".  The third form is for continuing the
+       definition of a nonterminal when it has multiple productions.  It leaves the first
+       column in the output blank.
 
     """
     subdomain = "prodn"
     #annotation = "Grammar production"
 
+    prev_position = None
+    table = None
+    tbody = None
+
     def _render_signature(self, signature, signode):
         raise NotImplementedError(self)
 
     SIG_ERROR = ("{}: Invalid syntax in ``.. prodn::`` directive"
                  + "\nExpected ``name ::= ...`` or ``name += ...``"
-                 + " (e.g. ``pattern += constr:(@ident)``)")
+                 + " (e.g. ``pattern += constr:(@ident)``)\n"
+                 + "  in `{}`")
 
     def handle_signature(self, signature, signode):
-        nsplits = 2
-        parts = signature.split(maxsplit=nsplits)
-        if len(parts) != 3:
-            loc = os.path.basename(get_node_location(signode))
-            raise ExtensionError(ProductionObject.SIG_ERROR.format(loc))
-
-        lhs, op, rhs = (part.strip() for part in parts)
-        if op not in ["::=", "+="]:
-            loc = os.path.basename(get_node_location(signode))
-            raise ExtensionError(ProductionObject.SIG_ERROR.format(loc))
-
-        self._render_annotation(signode)
-
-        lhs_op = '{} {} '.format(lhs, op)
-        lhs_node = nodes.literal(lhs_op, lhs_op)
+        parts2 = signature.split(maxsplit=1)
+        if parts2[0].strip() == "|":
+            lhs = ""
+            op = "|"
+            rhs = parts2[1].strip()
+        else:
+            nsplits = 2
+            parts = signature.split(maxsplit=nsplits)
+            if len(parts) != 3:
+                loc = os.path.basename(get_node_location(signode))
+                raise ExtensionError(ProductionObject.SIG_ERROR.format(loc, signature))
+            else:
+                lhs, op, rhs = (part.strip() for part in parts)
+                if op not in ["::=", "+="]:
+                    loc = os.path.basename(get_node_location(signode))
+                    raise ExtensionError(ProductionObject.SIG_ERROR.format(loc, signature))
 
         position = self.state_machine.get_source_and_line(self.lineno)
+        if lhs == '':
+            lhs_node = nodes.literal('', '')
+        else:
+            lhs_node = notation_to_sphinx('@'+lhs, *position)
         rhs_node = notation_to_sphinx(rhs, *position)
-        signode += addnodes.desc_name(signature, '', lhs_node, rhs_node)
+
+        # create a row
+        row = nodes.row()
+        entry = nodes.entry(classes=['prodn_col1'])
```

Comment:
I think a name like `prodn_nonterminal` would probably be better than `prodn_col1`

Reply:
OK.  Hard to think of a good name for the middle column, though.